File: java/commonWeb/src/main/java/web/template/controller/common/LogEntityController.java
package web.template.controller.common;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.txj.common.entity.Result;

import web.template.params.LogEntityParams;
import web.template.service.common.LogEntityService;

/**
 * “日志管理”模块的控制器类
 * 
 * @author admin
 */
@RestController
@RequestMapping("/LogEntity")
public class LogEntityController extends BaseController {

	@Autowired
	private LogEntityService service;

	/**
	 * 分页查询“日志管理”模块，并返回查询结果
	 * 
	 * @param param
	 *            查询参数
	 * @param currentPageIndex
	 *            当前页码
	 * @param pageSize
	 *            每页显示的数据量
	 * @return 返回“日志管理”模块的查询结果
	 */
	@RequestMapping("/page")
	public Result page(LogEntityParams param, int currentPageIndex, int pageSize) {
		log.error("测试日志", new RuntimeException("致命错误"));
		return new Result(0, null, service.page(param, currentPageIndex, pageSize));
	}

	/**
	 * 根据主键id查询“日志管理”模块的数据实体
	 * 
	 * @param id
	 *            主键id
	 * @param return
	 *            返回“日志管理”模块的查询结果
	 */
	@RequestMapping("/load")
	public Result load(Long id) {
		return new Result(0, null, service.load(id));
	}

	/*
	 * 抄考代码
	 * -------------------------------------------------------------------------
	 * --------------------------------------------------- LogEntity data = new
	 * LogEntity();
	 * 
	 * data.setId(nextId());
	 * 
	 * data.setCreateDate(param.getCreateDate());
	 * 
	 * 
	 * 
	 * data.setLevel(param.getLevel());
	 * 
	 * 
	 * data.setThreadNo(param.getThreadNo());
	 * 
	 * 
	 * data.setMessage(param.getMessage());
	 * 
	 * 
	 * data.setProjectName(param.getProjectName());
	 * 
	 * 
	 * data.setTypeName(param.getTypeName());
	 * 
	 * 
	 * data.setFuncName(param.getFuncName());
	 * 
	 * 
	 * data.setException(param.getException());
	 * 
	 * 
	 * -------------------------------------------------------------------------
	 * --------------------------------------------------- LogEntityParams param
	 * = new LogEntityParams();
	 * 
	 * param.setId(param.getId());
	 * 
	 * param.setCreateDate(param.getCreateDate());
	 * 
	 * param.setCreateDateStart(param.getCreateDateStart());
	 * 
	 * param.setCreateDateEnd(param.getCreateDateEnd());
	 * 
	 * param.setLevel(param.getLevel());
	 * 
	 * param.setLevelLike(param.getLevelLike());
	 * 
	 * param.setThreadNo(param.getThreadNo());
	 * 
	 * param.setThreadNoLike(param.getThreadNoLike());
	 * 
	 * param.setMessage(param.getMessage());
	 * 
	 * param.setMessageLike(param.getMessageLike());
	 * 
	 * param.setProjectName(param.getProjectName());
	 * 
	 * param.setProjectNameLike(param.getProjectNameLike());
	 * 
	 * param.setTypeName(param.getTypeName());
	 * 
	 * param.setTypeNameLike(param.getTypeNameLike());
	 * 
	 * param.setFuncName(param.getFuncName());
	 * 
	 * param.setFuncNameLike(param.getFuncNameLike());
	 * 
	 * param.setException(param.getException());
	 * 
	 * param.setExceptionLike(param.getExceptionLike());
	 * 
	 * -------------------------------------------------------------------------
	 * --------------------------------------------------- 分页查询***模块，并返回查询结果
	 * 
	 * @param param 查询参数
	 * 
	 * @param currentPageIndex 当前页码
	 * 
	 * @param pageSize 每页显示的数据量
	 * 
	 * @return 返回***模块的查询结果
	 * 
	 * @RequestMapping("/page") public Result page(LogEntityParams param,int
	 * currentPageIndex,int pageSize){ return new
	 * Result(0,null,service.page(param, currentPageIndex, pageSize)); }
	 * -------------------------------------------------------------------------
	 * --------------------------------------------------- 导出***模块，使用excel表保存
	 * 
	 * @param param 筛选参数
	 * 
	 * @param excelType excel的后缀名、类型
	 * 
	 * @param currentPageIndex 当前页码
	 * 
	 * @param pageSize 每页显示的数据量
	 * 
	 * @return 返回excel
	 * 
	 * @RequestMapping("/export") public ModelAndView export(LogEntityParams
	 * param, String excelType, int currentPageIndex, int pageSize){
	 * Map<String,Object> model=new HashMap<String,Object>();
	 * model.put(MyExcelView.EXCEL_DATA, service.export(param, currentPageIndex,
	 * pageSize).getPageDataList()); model.put(MyExcelView.EXCEL_NAME,
	 * "excel文件名称."+excelType); model.put(MyExcelView.GROUP_NAME, "组名称"); return
	 * new ModelAndView(myExcelView, model); }
	 * -------------------------------------------------------------------------
	 * ---------------------------------------------------
	 * 
	 * @RequestMapping("/import") public Result
	 * importExcel(@RequestParam("fileUpload")MultipartFile fileUpload,
	 * 其他参数...){ List<LogEntity> LogEntityList; if
	 * (fileUpload.getOriginalFilename().endsWith("xlsx")) { LogEntityList =
	 * ExcelHelper.excelXlsxToList<LogEntity>(fileUpload.InputStream); } else if
	 * (fileUpload.getOriginalFilename().endsWith("xls")) { LogEntityList =
	 * ExcelHelper.excelXlsToList<LogEntity>(fileUpload.InputStream); }
	 * service.addBatch(LogEntityList); return new Result(0, "导入成功。",null); }
	 * -------------------------------------------------------------------------
	 * ---------------------------------------------------
	 * 
	 * @RequestMapping("/del") public Result del(long id){ service.del(id);
	 * return new Result(0, "数据已删除。",null); }
	 * -------------------------------------------------------------------------
	 * --------------------------------------------------- 新增一条数据
	 * 
	 * @param data 新增的数据
	 * 
	 * @RequestMapping("/add") public Result add(LogEntity data){
	 * service.add(data); return new Result(0, "保存成功。",null); }
	 * -------------------------------------------------------------------------
	 * --------------------------------------------------- 批量修改状态
	 * 
	 * @param datas 修改状态的数据
	 * 
	 * @RequestMapping("/changeStatus") public Result changeStatus(LogEntity
	 * datas){ return new
	 * Result(0,"修改成功，共"+service.changeStatus(datas)+"条。",null); }
	 * -------------------------------------------------------------------------
	 * --------------------------------------------------- 根据主键id查询***模块的数据实体
	 * 
	 * @param id 主键id
	 * 
	 * @param return 返回***模块的查询结果
	 * 
	 * @RequestMapping("/load") public Result load(Long id){ return new
	 * Result(0,null ,service.Load(id) ); }
	 * -------------------------------------------------------------------------
	 * --------------------------------------------------- 批量删除***模块的数据实体
	 * 
	 * @param datas 删除的数据主键列表
	 * 
	 * @RequestMapping("/delBatch") public Result delBatch(List<LogEntity>
	 * datas){ return new Result(0 ,null,"删除成功，共"+Service.DelBatch(datas)+"条。");
	 * }
	 */
}

File: c#/CommonWeb/WebApplication1/Controllers/LogEntityController.cs
using CommonHelper.CommonEntity;
using CommonHelper.Helper.CommonEntity;
using CommonWeb.Controllers.Common;
using CommonWeb.Filter.Common;
using CommonWeb.Service;
using System;
using System.Collections.Generic;
using System.Data.Entity;
using System.Linq;
using System.Web;
using System.Web.Mvc;
namespace CommonWeb.Controllers
{
    /// <summary>
    /// “日志管理”模块的控制器类
    /// </summary>
    public sealed partial class LogEntityController : BaseController
    {
        public LogEntityService Service { set; get; }

        /// <summary>
        /// 分页查询“日志管理”模块，并返回查询结果
        /// </summary>
        /// <param name="param">查询参数</param>
        /// <param name="currentPageIndex">当前页码</param>
        /// <param name="pageSize">每页显示的数据量</param>
        /// <returns>返回“日志管理”模块的查询结果</returns>
		[Compress]
        public JsonResult Page(LogEntityParams param, int currentPageIndex = 1, int pageSize = 20)
        {
            return MyJson(new Result { code = 0, data = Service.Page(currentPageIndex, pageSize, param) });
        }

        /// <summary>
        /// 根据主键id查询“日志管理”模块的数据实体
        /// </summary>
        /// <param name="id">主键id</param>
		/// <returns>返回“日志管理”模块的查询结果</returns>
		public JsonResult Load(long id)
        {
            return MyJson(new Result { code = 0, data = Service.Load(id) });
        }

        /*抄考代码
----------------------------------------------------------------------------------------------------------------------------
		LogEntity data = new LogEntity
		{

					ID = Next(),
		
					CreateDate = param.CreateDate,
		
		
		
					Level = param.Level,
		
		
					ThreadNo = param.ThreadNo,
		
		
					Message = param.Message,
		
		
					Namespace = param.Namespace,
		
		
					TypeName = param.TypeName,
		
		
					MethodName = param.MethodName,
		
		
					USERNAME = param.USERNAME,
		
		
		};
----------------------------------------------------------------------------------------------------------------------------
		LogEntityParams param = new LogEntityParams
		{

			ID = param.ID,
		
			CreateDate = param.CreateDate,
		
			CreateDateStart = param.CreateDateStart,
		
			CreateDateEnd = param.CreateDateEnd,
		
			Level = param.Level,
		
			LevelLike = param.LevelLike,
		
			ThreadNo = param.ThreadNo,
		
			ThreadNoLike = param.ThreadNoLike,
		
			Message = param.Message,
		
			MessageLike = param.MessageLike,
		
			Namespace = param.Namespace,
		
			NamespaceLike = param.NamespaceLike,
		
			TypeName = param.TypeName,
		
			TypeNameLike = param.TypeNameLike,
		
			MethodName = param.MethodName,
		
			MethodNameLike = param.MethodNameLike,
		
			USERNAME = param.USERNAME,
		
			USERNAMELike = param.USERNAMELike,
		
		};
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 分页查询***模块，并返回查询结果
        /// </summary>
        /// <param name="param">查询参数</param>
        /// <param name="currentPageIndex">当前页码</param>
        /// <param name="pageSize">每页显示的数据量</param>
        /// <returns>返回***模块的查询结果</returns>
		[Compress]
		public JsonResult Page(LogEntityParams param,int currentPageIndex = 1,int pageSize = 20)
		{
			return MyJson(new Result{code = 0, data = Service.Page(param, currentPageIndex, pageSize)});
		}
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 导出***模块，使用excel表保存
        /// </summary>
        /// <param name="param">查询参数</param>
		/// <param name="excelType">excel类型</param>
        /// <param name="currentPageIndex">当前页码</param>
        /// <param name="pageSize">每页显示的数据量</param>
        /// <returns>返回***模块的导出结果</returns>
		[OperInterval(IntervalMillisecond=10000)]
		[Compress]
		public ExcelResult<LogEntity> Export(LogEntityParams param, string excelType, int currentPageIndex = 1, int pageSize = 10000)
		{
			return new ExcelResult<LogEntity>
			{
				DataList = Service.Page(param, currentPageIndex, pageSize).pageDataList,
				FileName = "测试excel."+excelType
			};
		}
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 导入***模块数据
        /// </summary>
        /// <param name="param">查询参数</param>
        /// <param name="currentPageIndex">当前页码</param>
        /// <param name="pageSize">每页显示的数据量</param>
        /// <returns>返回***模块的导出结果</returns>
		public JsonResult Import(HttpPostedFileBase fileUpload, 其他参数...)
		{
			List<LogEntity> LogEntityList;
            if (fileUpload.FileName.EndsWith("xlsx"))
            {
                LogEntityList = ExcelHelper.ExcelXlsxToList<LogEntity>(fileUpload.InputStream);
            }
            else if (fileUpload.FileName.EndsWith("xls"))
            {
                LogEntityList = ExcelHelper.ExcelXlsToList<LogEntity>(fileUpload.InputStream);
            }
			Service.AddBatch(LogEntityList);
			return MyJson(new Result { code = 0, msg = "导入成功。"});
		}
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 根据主键删除指定数据
        /// </summary>
        /// <param name="ID">删除数据的主键</param>
		public JsonResult Del(long ID)
		{
            Service.Del(ID);
			return MyJson(new Result { code = 0, msg = "数据已删除。"});
		}
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 新增一条数据
        /// </summary>
        /// <param name="data">新增的数据</param>
		public JsonResult Add(LogEntity data)
		{
			Service.Add(data);
			return MyJson(new Result { code = 0, msg = "保存成功。"});
		}
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 批量修改状态
        /// </summary>
        /// <param name="datas">修改状态的数据</param>
		public JsonResult ChangeStatus(LogEntity datas)
		{
			return MyJson(new Result { code = 0,msg=$"修改成功，共{Service.ChangeStatus(datas)}条。"});
		}
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 根据主键id查询***模块的数据实体
        /// </summary>
        /// <param name="ID">主键id</param>
		/// <returns>返回***模块的查询结果</returns>
		public JsonResult Load(long ID)
		{
			return MyJson(new Result { code = 0, data = Service.Load(ID) });
		}
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 批量删除数据
        /// </summary>
        /// <param name="datas">批量删除的数据</param>
		public JsonResult DelBatch(List<LogEntity> datas)
		{
			return MyJson(new Result { code = 0 ,msg = $"删除成功，共{Service.DelBatch(datas)}条。" });
		}
		*/
    }
}